backend/src/routes/user.ts

```diff
@@ -1,12 +1,47 @@
-import { Request, Response, Router } from 'express';
+import { NextFunction, Request, Response, Router } from 'express';
+import { BAD_REQUEST } from 'http-status';
+import { makeError } from 'helpers/error';
 import User from 'models/user';
 
 const router = Router();
 
-router.get('/', async (req: Request, res: Response) => {
-  const users = await User.findAll();
+router.get('/', async (req: Request, res: Response, next: NextFunction) => {
+  await User.findAll()
+    .then(users => res.json(users))
+    .catch(error => next(error));
+});
+
+router.put('/', async (req: Request, res: Response, next: NextFunction) => {
+  const { id } = req.user as User;
```

Comment:
This casting to `User` type is done a lot in this file. Can it be avoided?

Reply:
Yes, I didn't test it but it should be possible. Fix should be adding our own definition for the request type, we can create the custom interface that extends the Request interface by adding the user field to it. That will be individual PR in the future.